docs/topics/gradle.md

```diff
@@ -133,6 +133,158 @@ sourceSets {
+
+When you use a custom JDK, note that [kapt task workers](kapt.md#running-kapt-tasks-in-parallel)
+use [process isolation mode](https://docs.gradle.org/current/userguide/worker_api.html#changing_the_isolation_mode) only,
+and ignore the `kapt.workers.isolation` property.
+
+As before, if you don't set the toolchain or the `jdkHome` option, the Kotlin/JVM compilation uses the current user's JDK.
+
+### Gradle Java toolchains support
+
+Gradle 6.7 introduced ["Java toolchains support"](https://docs.gradle.org/current/userguide/toolchains.html).
+Using this feature, you can:
+* Use a JDK and a JRE that are different from the Gradle ones to run compilations, tests, and executables.
+* Compile and test code with a not-yet-released language version.
+
+With toolchains support, Gradle can autodetect local JDKs and install missing JDKs that Gradle requires for the build.
+Now Gradle itself can run on any JDK and still reuse the [remote build cache feature](#gradle-build-cache-support)
+for tasks that depend on a major JDK version.
+
+The Kotlin Gradle plugin supports Java toolchains for Kotlin/JVM compilation tasks. JS and Native tasks don't use toolchains.
+The Kotlin compiler always uses the JDK the Gradle daemon is running on.
+A Java toolchain:
+* Sets the [`jdkHome` option](#attributes-specific-to-jvm) available for JVM targets.
+* Sets the [`kotlinOptions.jvmTarget`](#attributes-specific-to-jvm) to the toolchain's JDK version
+  if the user doesn't set the `jvmTarget` option explicitly.
+  If the user doesn't configure the toolchain, the `jvmTarget` field will use the default value.
+  Learn more about [JVM target compatibility](#check-for-jvm-target-compatibility-of-related-compile-tasks).
+* Affects which JDK [`kapt` workers](kapt.md#running-kapt-tasks-in-parallel) are running on.
+
+Use the following code to set a toolchain. Replace the placeholder `<MAJOR_JDK_VERSION>` with the JDK version you would like to use:
+
+<tabs group="build-script">
+<tab title="Kotlin" group-key="kotlin">
+
+```kotlin
+kotlin {
+    jvmToolchain {
+        (this as JavaToolchainSpec).languageVersion.set(JavaLanguageVersion.of(<MAJOR_JDK_VERSION>)) // "8" 
+    }
+}
+```
+
+
+</tab>
+<tab title="Groovy" group-key="groovy">
+
+```groovy
+kotlin {
+    jvmToolchain {
+        languageVersion.set(JavaLanguageVersion.of(<MAJOR_JDK_VERSION>)) // "8"
+    }
+}
+```
+
+</tab>
+</tabs>
+
+Note that setting a toolchain via the `kotlin` extension will update the toolchain for Java compile tasks as well.
+
+You can set a toolchain via the `java` extension, and Kotlin compilation tasks will use it:
+
+```kotlin
+java {
+    toolchain {
+      languageVersion.set(JavaLanguageVersion.of(<MAJOR_JDK_VERSION>)) // "8" 
+    }
+}
+```
+
+To set any JDK (even local) for the specific task, use the Task DSL.
+
+### Setting JDK version with the Task DSL
```

Comment:
This subsection needs an intro describing the case - older Gradle without toolchains.
Otherwise readers that just scan headings with their eyes or Ctrl+F "JDK version" may get there and think that this is a general way.